### ISSUE TITLE
[gdb][Linux] Sudo elevation not working with VSCode RunInTerminal

### ISSUE THREAD

**Issue by princehektor** [microsoft/vscode-cpptools#2753]
Type: Debugger
**Input information below**

**Please review existing issues and our documentation at https://github.com/Microsoft/vscode-cpptools/tree/master/Documentation prior to filing an issue.**

**Describe the bug**

OS and Version: Ubuntu 18, 64bit
VS Code Version: 1.28.2
C/C++ Extension Version: 0.20.0
Other extensions you installed (and if the issue persists after disabling them):

The debugger in VS code does not work anymore after updating to Ubuntu 18 and also updating VS Code to the latest version.

Earlier behavior:

Starting the debugger would open xterm and prompt me my for my password and then i would be good to debug my application.

Current behavior:

Starting the debugger seems to be opening up the xfce4 terminal with the following output.

[1] + Stopped (tty output)       /usr/bin/gdb --interpreter=mi --tty=${DbgTerm} 0</tmp/Microsoft-MIEngine-In-ongysklf.4yq 1>/tmp/Microsoft-MIEngine-Out-ybu9kjug.08a
You have stopped jobs.

Press any key to continue...[sudo] 

If i press any key to continue, the terminal just closes.

Anyone see the same issue ?

- A clear and concise description of what the bug is.

**To Reproduce**
*Please include a code sample and `launch.json` configuration.*

**Additional context**
*If applicable, please include logging by adding "logging": { "engineLogging": true, "trace": true, "traceResponse": true } in your `launch.json`*
Add any other context about the problem here including log or error messages in your Debug Console or Output windows.

**Comment by princehektor** [comment]
@sean-mcmanus , @pieandcakes : Could you guys help me with this ?

**Comment by princehektor** [comment]
I rebooted my machine and tried debugging from VSCode. It still opens up a terminal with a different message now.

nied

[1] + Done(126)                  /usr/bin/sgdb --interpreter=mi --tty=${DbgTerm} 0</tmp/Microsoft-MIEngine-In-dvq9gn3l.eth 1>/tmp/Microsoft-MIEngine-Out-p0fhame1.28t

Press any key to continue...

This is my launch.json


        {
            "name": "(gdb) my-debug",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceFolder}/myproj,
            "args": [],
            "stopAtEntry": true,
            "cwd": "${workspaceFolder}/myproj",
            "externalConsole": true,
            "MIMode": "gdb",
            "miDebuggerPath": "/usr/bin/sgdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                },
                {
                    "description": "Have gdb ignore SIGUSR1",
                    "text": "handle SIGUSR1 nostop noprint",
                    "ignoreFailures": true
                },
                {
                    "description": "Have gdb ignore SIGUSR2",
                    "text": "handle SIGUSR2 nostop noprint",
                    "ignoreFailures": true
                }
            ]
        },

sgdb is simply a script which opens gdb using sudo. 

I see the same behavior if i attempt to run vs code using sudo.

**Comment by pieandcakes** [comment]
We changed over to using VSCode's debugProtocol to handle terminal launching. This is now using the RunInTerminalRequest message for VS Code. Can you try changing `externalConsole` to false and see if it works? I don't know if they are blocking any other UI from spawning.

**Comment by princehektor** [comment]
This is the message i see in the integrated terminal after setting "externalConsole" to false.

[1] + Done(126)                  /usr/bin/sgdb --interpreter=mi --tty=${DbgTerm} 0</tmp/Microsoft-MIEngine-In-2zw8gvpe.arx 1>/tmp/Microsoft-MIEngine-Out-y60x1n13.mis


Is there any way for me to revert to the old VSCode behavior ? Will downgrading to a previous version help ?


Update : I tried downgrading to version 1.26 but still see the same behavior. I dont think it is a problem with the Ubuntu upgrade to 18.0.4. One of my friends is seeing the same problem with Ubuntu 16.04 as well. Everything used to work but he recently did a "sudo apt-get update" and "sudo apt-get upgrade" and also upgraded VSCode to the latest version. That is when things stopped working for him.

**Comment by pieandcakes** [comment]
we made the change in 0.20.0 so anything before that has the old behavior. 

To be clear, this is the C++ extension that made the transition due to customer requests. You will need to roll back our extension and not VS Code.

**Comment by princehektor** [comment]
Okay. How do i do that ?

Where can i find the ".vsix" file for version 0.19.0 ?

**Comment by princehektor** [comment]
I managed to get the ".vsix" file for the older version from here (https://github.com/Microsoft/vscode-cpptools/releases)

I tried to install it and it warned me that a newer version is avaiable and whether i still wanted to over-ride it. I chose yes and VSCode prompted me to reload the windows which i did.

Now when i try to debug, it says :

"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information"

Closing and opening the workspace brings the extension back to version 0.20.1 .

So, what is the correct way to downgrade the C++ extension to version 0.19.0 ?

**Comment by pieandcakes** [comment]
* Uninstall the current version. 
* Reload VS Code
* Go to File -> Preferences -> Settings and disable Extension: Auto Update
![image](https://user-images.githubusercontent.com/17558378/47874137-3b24be80-ddd0-11e8-88c8-10d85cd0f09c.png)
* Install the older version of the .vsix

**Comment by princehektor** [comment]
Did that but still see this error :

"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information"

**Comment by princehektor** [comment]
My bad. My sgdb script did not have execute permissions.

@pieandcakes : So it is confirmed that this issue does not exist in version 0.19 of the C++ extension.

Is there anyway you guys can add a new setting to toggle this new 'VSCode's debugProtocol behavior' that was introduced in version 0.20 and beyond ? This way users can toggle it as needed.

Maybe add that to the next version of the C/C++ extension ?

What say ?

**Comment by pieandcakes** [comment]
@princehektor the code is in [MIEngine](https://github.com/Microsoft.MIEngine) although we removed the old behavior. I think the right way would be to either A) figure out what the issue is and have VS Code fix it or B) figure out a workaround for the problem that works. 

This is on my list of items to investigate but I don't have time currently. If you want to include some logging and see if you can repro it in a VS Code console window, we can file a bug with them to fix it.

**Comment by princehektor** [comment]
The workaround for me currently is to downgrade to version 0.19 of the extension. But with this i am missing out on all the fixes and improvements in later versions.

So the workaround is okay for now but i would prefer a better long term solution.

How do i enable logging and get you the information that you needed? Please help me out here.

**Comment by pieandcakes** [comment]
Instructions are located in our [FAQ](https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/FAQs.md#how-to-enable-logging) Specifically, you probably want `trace` and `traceResponse` set to see what we send to VS Code and what you get back.

**Comment by princehektor** [comment]
@pieandcakes :This is the output from the debug console with trace, traceResponse and enableLogging set to to `true`. Let me know if this is what you wanted.

--> E (output): {"event":"output","body":{"category":"console","output":"1: (93) LaunchOptions<LocalLaunchOptions xmlns='http://schemas.microsoft.com/vstudio/MDDDebuggerOptions/2014'\n"},"seq":2,"type":"event"}
1: (93) LaunchOptions<LocalLaunchOptions xmlns='http://schemas.microsoft.com/vstudio/MDDDebuggerOptions/2014'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (100) LaunchOptions  ExePath='my_proj_path'\n"},"seq":4,"type":"event"}
1: (100) LaunchOptions  ExePath='my_proj_path'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (100) LaunchOptions  WorkingDirectory='my_proj_path'\n"},"seq":6,"type":"event"}
1: (100) LaunchOptions  WorkingDirectory='my_proj_path'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (100) LaunchOptions  ExeArguments=''\n"},"seq":8,"type":"event"}
1: (100) LaunchOptions  ExeArguments=''
--> E (output): {"event":"output","body":{"category":"console","output":"1: (100) LaunchOptions  MIMode='gdb'\n"},"seq":10,"type":"event"}
1: (100) LaunchOptions  MIMode='gdb'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (101) LaunchOptions  MIDebuggerPath='/usr/bin/sgdb'\n"},"seq":12,"type":"event"}
1: (101) LaunchOptions  MIDebuggerPath='/usr/bin/sgdb'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (101) LaunchOptions  WaitDynamicLibLoad='false'\n"},"seq":14,"type":"event"}
1: (101) LaunchOptions  WaitDynamicLibLoad='false'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (101) LaunchOptions  ExternalConsole='true'\n"},"seq":16,"type":"event"}
1: (101) LaunchOptions  ExternalConsole='true'
--> E (output): {"event":"output","body":{"category":"console","output":"1: (101) LaunchOptions>\n"},"seq":18,"type":"event"}
1: (101) LaunchOptions>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (102) LaunchOptions    <SetupCommands>\n"},"seq":20,"type":"event"}
1: (102) LaunchOptions    <SetupCommands>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (102) LaunchOptions        <Command IgnoreFailures='true' Description='Enable pretty-printing for gdb'>-enable-pretty-printing</Command>\n"},"seq":22,"type":"event"}
1: (102) LaunchOptions        <Command IgnoreFailures='true' Description='Enable pretty-printing for gdb'>-enable-pretty-printing</Command>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (102) LaunchOptions        <Command IgnoreFailures='true' Description='Have gdb ignore SIGUSR1'>handle SIGUSR1 nostop noprint</Command>\n"},"seq":24,"type":"event"}
1: (102) LaunchOptions        <Command IgnoreFailures='true' Description='Have gdb ignore SIGUSR1'>handle SIGUSR1 nostop noprint</Command>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (102) LaunchOptions        <Command IgnoreFailures='true' Description='Have gdb ignore SIGUSR2'>handle SIGUSR2 nostop noprint</Command>\n"},"seq":26,"type":"event"}
1: (102) LaunchOptions        <Command IgnoreFailures='true' Description='Have gdb ignore SIGUSR2'>handle SIGUSR2 nostop noprint</Command>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (103) LaunchOptions    </SetupCommands>\n"},"seq":28,"type":"event"}
1: (103) LaunchOptions    </SetupCommands>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (103) LaunchOptions</LocalLaunchOptions>\n"},"seq":30,"type":"event"}
1: (103) LaunchOptions</LocalLaunchOptions>
--> E (output): {"event":"output","body":{"category":"console","output":"1: (193) DbgCmd:echo $$ > /tmp/Microsoft-MIEngine-Pid-cmtgz3u3.nin ; cd my_proj_path ; DbgTerm=`tty` ; set -o monitor ; trap 'rm /tmp/Microsoft-MIEngine-In-lv16n8e0.bi5 /tmp/Microsoft-MIEngine-Out-9lkjodby.rqn /tmp/Microsoft-MIEngine-Pid-cmtgz3u3.nin /tmp/Microsoft-MIEngine-Cmd-7jpdxvw9.jzv' EXIT ; /usr/bin/sgdb --interpreter=mi --tty=$DbgTerm < /tmp/Microsoft-MIEngine-In-lv16n8e0.bi5 > /tmp/Microsoft-MIEngine-Out-9lkjodby.rqn & clear; pid=$! ; echo $pid > /tmp/Microsoft-MIEngine-Pid-cmtgz3u3.nin ; wait $pid; \n"},"seq":32,"type":"event"}
1: (193) DbgCmd:echo $$ > /tmp/Microsoft-MIEngine-Pid-cmtgz3u3.nin ; cd my_proj_path ; DbgTerm=`tty` ; set -o monitor ; trap 'rm /tmp/Microsoft-MIEngine-In-lv16n8e0.bi5 /tmp/Microsoft-MIEngine-Out-9lkjodby.rqn /tmp/Microsoft-MIEngine-Pid-cmtgz3u3.nin /tmp/Microsoft-MIEngine-Cmd-7jpdxvw9.jzv' EXIT ; /usr/bin/sgdb --interpreter=mi --tty=$DbgTerm < /tmp/Microsoft-MIEngine-In-lv16n8e0.bi5 > /tmp/Microsoft-MIEngine-Out-9lkjodby.rqn & clear; pid=$! ; echo $pid > /tmp/Microsoft-MIEngine-Pid-cmtgz3u3.nin ; wait $pid; 
--> C (runInTerminal-34): {"command":"runInTerminal","arguments":{"kind":"external","title":"cppdbg: foo","cwd":"","args":["sh","/tmp/Microsoft-MIEngine-Cmd-7jpdxvw9.jzv"],"env":{}},"seq":34,"type":"request"}
--> E (output): {"event":"output","body":{"category":"console","output":"1: (203) Wait for connection completion.\n"},"seq":36,"type":"event"}
1: (203) Wait for connection completion.

**Comment by pieandcakes** [comment]
Thanks. If you go to the terminal window in VS Code and just try and run `/usr/bin/sgdb --interpreter=mi` what do you get?

**Comment by princehektor** [comment]
/usr/bin/sgdb --interpreter=mi
=thread-group-added,id="i1"
~"GNU gdb (Ubuntu 8.1-0ubuntu3) 8.1.0.20180409-git\n"
~"Copyright (C) 2018 Free Software Foundation, Inc.\n"
~"License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.  Type \"show copying\"\nand \"show warranty\" for details.\n"~"This GDB was configured as \"x86_64-linux-gnu\".\nType \"show configuration\" for configuration details."~"\nFor bug reporting instructions, please see:\n"
~"<http://www.gnu.org/software/gdb/bugs/>.\n"
~"Find the GDB manual and other documentation resources online at:\n<http://www.gnu.org/software/gdb/documentation/>.\n"~"For help, type \"help\".\n"
~"Type \"apropos word\" to search for commands related to \"word\".\n"
(gdb)

**Comment by princehektor** [comment]
/usr/bin/sgdb --interpreter=mi
=thread-group-added,id="i1"
~"GNU gdb (Ubuntu 8.1-0ubuntu3) 8.1.0.20180409-git\n"
~"Copyright (C) 2018 Free Software Foundation, Inc.\n"
~"License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.  Type \"show copying\"\nand \"show warranty\" for details.\n"~"This GDB was configured as \"x86_64-linux-gnu\".\nType \"show configuration\" for configuration details."~"\nFor bug reporting instructions, please see:\n"
~"<http://www.gnu.org/software/gdb/bugs/>.\n"
~"Find the GDB manual and other documentation resources online at:\n<http://www.gnu.org/software/gdb/documentation/>.\n"~"For help, type \"help\".\n"
~"Type \"apropos word\" to search for commands related to \"word\".\n"
(gdb)

**Comment by pieandcakes** [comment]
so does that ask you for a password in a pop up box?

**Comment by princehektor** [comment]
Not in a pop up box. I was asked for the password in the integrated terminal itself.

**Comment by mjohngreene** [comment]
FYI - I am getting the exact same error:
"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information" on Windows 10.  

A colleague with an extremely similar desktop/environment to mine is using the same launch.json and tasks.json without error.

**Comment by princehektor** [comment]
@ssctx66 : I am guessing that your colleague probably hasnt reloaded VSCode after the C/C++ plugin update.I could be wrong though.

For me, i got the error when my sgdb file did not have execute permissions which i fixed with a chmod.

**Comment by mjohngreene** [comment]
Sorry, @princehektor, but I don't speak linux terribly well.  

My colleague has left for the day but he did confirm that he was using v0.20.1 (as am I).

The last few lines of my "Debug Console" output:

```
1: (922) Starting: "C:\openOCD\openocd-0.10.0\bin\openocd.exe" -f C:/OpenOCD/openocd-0.10.0/scripts/board/st_nucleo_f4.cfg -f C:/OpenOCD/openocd-0.10.0/scripts/interface/stlink-v2-1.cfg -c init -c "reset init"
1: (2275) Wait for connection completion.
Starting: "C:\openOCD\openocd-0.10.0\bin\openocd.exe" -f C:/OpenOCD/openocd-0.10.0/scripts/board/st_nucleo_f4.cfg -f C:/OpenOCD/openocd-0.10.0/scripts/interface/stlink-v2-1.cfg -c init -c "reset init"

1: (7334) <-logout
```
This output is the same as before up until the last line.   It's as though the GDB server is unable to launch.  I can launch GDB from command prompt/terminal, though.

**Comment by pieandcakes** [comment]
@ssctx66 Yours is on Windows which is a different implementation of what our problem is here on linux. Can you open a new issue with your `launch.json` so I can understand what it is you are passing?

**Comment by pieandcakes** [comment]
@princehektor I think I have a workaround for you.

I created a new script called `/usr/bin/sgdb` which has the following line in it:
```
pkexec /usr/bin/gdb $*
```

and then it works (it pops up a window asking for credentials). Will this work for you?

**Comment by princehektor** [comment]
@pieandcakes : So this is what i have in my gdb file right now.

#!/bin/sh
sudo gdb $@

You want me to add the line(pkexec /usr/bin/gdb $*)  at the end or instead of the sudo gdb line?

**Comment by pieandcakes** [comment]
@princehektor Yes please. When I did that, it worked for me. I think instead of invoking `sudo` use `pkexec` and it should work for you. It is just another program to prompt for elevation and I think its a default on Ubuntu.

**Comment by princehektor** [comment]
Okay i tried that. My sgdb file looks like this now

#!/bin/sh
pkexec /usr/bin/gdb $*

Like you mentioned, it opens up a terminal and prompts for my password. But after i enter my password, i get this message :

"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information"

This is the output from the debug console.

1: (127) LaunchOptions<LocalLaunchOptions xmlns='http://schemas.microsoft.com/vstudio/MDDDebuggerOptions/2014'
1: (136) LaunchOptions  ExePath='myproj'
1: (136) LaunchOptions  WorkingDirectory='myproj'
1: (136) LaunchOptions  ExeArguments=''
1: (136) LaunchOptions MIMode='gdb'
1: (136) LaunchOptions  MIDebuggerPath='/usr/bin/sgdb'
1: (136) LaunchOptions  WaitDynamicLibLoad='false'
1: (136) LaunchOptions  ExternalConsole='true'
1: (136) LaunchOptions>
1: (136) LaunchOptions    <SetupCommands>
1: (136) LaunchOptions        <Command IgnoreFailures='true' Description='Enable pretty-printing for gdb'>-enable-pretty-printing</Command>
1: (136) LaunchOptions        <Command IgnoreFailures='true' Description='Haanandve gdb ignore SIGUSR1'>handle SIGUSR1 nostop noprint</Command>
1: (136) LaunchOptions        <Command IgnoreFailures='true' Description='Have gdb ignore SIGUSR2'>handle SIGUSR2 nostop noprint</Command>
1: (136) LaunchOptions    </SetupCommands>
1: (136) LaunchOptions</LocalLaunchOptions>
1: (203) TempFile=/tmp/Microsoft-MIEngine-fifo-pzqxu36x.f8f
1: (203) TempFile=/tmp/Microsoft-MIEngine-fifo-eic9zfvf.och
1: (204) TempFile=/tmp/Microsoft-MIEngine-fifo-34ecn3c7.unn
1: (260) ShellPid=14030
1: (265) DebuggerPid=14033
1: (5666) Shell exited, stop debugging
E output: {"category":"stderr","output":"\r\n\n","data":null,"type":"output"}

E output: {"category":"telemetry","output":"VS/Diagnostics/Debugger/MIEngine/DebuggerAborted","data":{"VS.Diagnostics.Debugger.MIEngine.DebuggerName":"GDB","VS.Diagnostics.Debugger.MIEngine.LastSentCommandName":""},"type":"output"}
 R: {"success":false,"message":"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information.","request_seq":2,"command":"launch","body":{"error":{"id":1104,"format":"Unable to start debugging. Unable to establish a connection to GDB. Debug output may contain more information.","variables":null}},"running":false,"refs":null,"seq":0,"type":"response"}

**Comment by pieandcakes** [comment]
@princehektor did you close a window? I'm seeing `Shell exited, stop debugging` where it thinks the shell window was closed? Can you try with `externalConsole` set to `false` ?

**Comment by princehektor** [comment]
@pieandcakes : My bad. I was trying out the new sgdb file changes without reloading VSCode. Therefore it will still using the old version of the C/C++ extension. 

Once i reloaded, i get the same message as before :

Password:  [1] + Stopped (tty output)       /usr/bin/sgdb --interpreter=mi --tty=${DbgTerm} 0</tmp/Microsoft-MIEngine-In-4g0gtl2r.fvg 1>/tmp/Microsoft-MIEngine-Out-877qnd71.t7p

It doesnt even give me the option to enter my password.

Same thing with externalConsole set to false.

**Comment by pieandcakes** [comment]
I don't know why I'm not seeing the same thing. my `/usr/bin/sgdb` is literally `pkexec /usr/bin/gdb $@` and I ran a simple hello world with elevation. It prompts me to elevate and I enter my password and it works. I've tried it with both `externalConsole` true and false. Can you try a simple C++ program?

**Comment by princehektor** [comment]
Same thing with a simple hello world program. It lets me enter the password in the terminal but then just shows me the "unable to start debugging" error. 

Again,  all this  is with the C/C++ extension version 0.19. With the latest version, it is the same old tty problem the external terminal just closes when i enter any key.

Why was this changed in version 0.20 and above for everyone ? I mean why not a setting that could be toggled ? Was it something for security purposes ?

**Comment by pieandcakes** [comment]
@princehektor If your `sudo` solution works with 0.19.0 don't change it. my suggestion for `pkexec` was with `0.20.1`. I didn't try it with 0.19.0. I don't understand how you see a tty issue with 0.20.1 when using pkexec since it should be popping up to ask for UI. 

>Why was this changed in version 0.20 and above for everyone ? I mean why not a setting that could be toggled ? Was it something for security purposes ?

We have been asked by VS Code to support their way of launching terminals and to conform to what other extensions do.

**Comment by princehektor** [comment]
@pieandcakes : Well that is unfortunate. For the time being i can continue using version 0.19.0 but i hope you guys can figure out a way to fix this so that we can get all the fixes you guys add with newer versions.

 More and more people are going to run into similar issues i feel.

**Comment by pieandcakes** [comment]
On my machine, with 0.20.1 I can use my `pkexec` solution and it works. If you want to dig into it, we can try and figure out why it doesn't work for you. 

I setup a clean Ubuntu 18.04 VSCode and the 0.20.1 version of the extension and it works fine too, so I don't know if it has to do with some other configuration you have set within VS Code.

**Comment by princehektor** [comment]
@pieandcakes : With version 0.20.1, it does open a new terminal and it looks like it is prompting for password but before i have a chance to enter the password, the tty problem hits.

Password: [1] + Stopped (tty output)       /usr/bin/sgdb --interpreter=mi --tty=${DbgTerm} 0</tmp/Microsoft-MIEngine-In-iwzc6aiq.feu 1>/tmp/Microsoft-MIEngine-Out-r36u9ljh.zrr
You have stopped jobs.

Press any key to continue...

I thought this is because i was using VNC to access my Ubuntu machine but it turns out the same thing happens when i try locally as well.

**Comment by pieandcakes** [comment]
Are you using a custom terminal instead of the default terminal in VS Code? That's the only other thing I can think of. Also, with `pkexec` in your sgdb file it shouldn't show the password message in your output. That's what confuses me.

If you can give me something to repro this I'll debug it, but on my clean Ubuntu machine it is working.

**Comment by princehektor** [comment]
Not using any custom terminal. I dont have gnome-terminal installed - only xterm and xfce terminal.

When i use version 0.19.0 and everything works, xterm opens up prompting for password and then i can continue on debugging. 

I will ask my other colleagues to try it out as well and let you know the results.

**Comment by princehektor** [comment]
@pieandcakes : I have some good news for you.

One of my colleagues updated the extension to version 0.20.1 and used "pkexec gdb" instead of "sudo gdb" in the sgdb file. Everything worked fine with no problems. He was also able to use the integrated terminal for debugging.

One of the differences between our setups is that my colleague was using the Ubuntu machine natively whereas i am using VNC to connect. So i tried the "pkexec" solution on my system locally (instead of VNC) and i was able to debug correctly. Only thing was that it showed this message on exit 

[1] + Done                       /usr/bin/sgdb --interpreter=mi --tty=${DbgTerm} 0</tmp/Microsoft-MIEngine-In-o61b60mq.5vu 1>/tmp/Microsoft-MIEngine-Out-z9nqak38.zue

Press any key to continue...

Once i pressed a key, the external console closed correctly. This behavior does not exist if i use the integrated terminal 

So it turns out that the problem is using my VNC setup. I can still use VNC and get everything to work if i open VSCode as sudo.

To summarize:

1. No problems when using the Ubuntu system locally.
2. No problems when opening VSCode using sudo. 

I will try to use x11vnc instead and see what happens.

**Comment by pieandcakes** [comment]
Thank you for investigating. I'll consider this resolved then.

**Comment by princehektor** [comment]
Yeah sure. Thanks a lot for your help with this. Much appreciated.

**Comment by sadeeq-bite** [comment]
![entirescreen](https://user-images.githubusercontent.com/82848932/119239451-2091cc80-bb17-11eb-9363-4265396aadd1.png)
